src/main/webapp/app/shared/util/firebase/firebase-reviewable-crud-store.spec.ts

```diff
@@ -0,0 +1,160 @@
+  };
+});
+
+const updateGeneMetaContentMock = jest.spyOn(FirebaseMetaStore.prototype, 'updateGeneMetaContent').mockReturnValue(Promise.resolve());
+const updateGeneReviewUuidMock = jest.spyOn(FirebaseMetaStore.prototype, 'updateGeneReviewUuid').mockReturnValue(Promise.resolve());
+
+describe('FirebaseReviewableCrudStore', () => {
+  class TestRootStore {
+    firebaseStore;
+    firebaseMetaStore;
+    authStore;
+    constructor() {
+      this.firebaseStore = new FirebaseStore(this as any);
+      this.firebaseMetaStore = new FirebaseMetaStore(this as any);
+      this.authStore = { fullName: 'test user' };
+    }
+  }
+
+  let rootStore = undefined;
+  const defaultDate = new Date('2023-01-01');
+
+  const reset = () => {
+    jest.useFakeTimers().setSystemTime(defaultDate);
+    rootStore = new TestRootStore();
+    jest.clearAllMocks();
+  };
+
+  describe('updateReviewableContent', () => {
+    beforeEach(() => reset());
+
+    it('should correctly update field that was already reviewed', async () => {
+      const store = new FirebaseReviewableCrudStore<any>(rootStore);
+      store.data = { key: 'original', key_uuid: 'fake-uuid' };
+
+      await store.updateReviewableContent('Gene/ABL1', 'key', 'new value');
+
+      expect(mockUpdate).toHaveBeenCalled();
+
+      // Check if field and field_review were updated
+      // lastReviewed should be set because the field was already reviewed and we are making a new change
+      const updateValueArg = mockUpdate.mock.calls[0][1];
+      expect(updateValueArg).toBeDefined();
+      expect(updateValueArg).toEqual({
+        key: 'new value',
+        key_review: { updateTime: defaultDate.getTime(), updatedBy: 'test user', lastReviewed: 'original' },
+      });
+
+      // Check if Meta information was updated
+      expect(updateGeneMetaContentMock).toHaveBeenCalled();
+      expect(updateGeneMetaContentMock.mock.calls[0][0]).toEqual('ABL1');
+
+      // Check if we add uuid to meta
+      expect(updateGeneReviewUuidMock).toHaveBeenCalled();
+      expect(updateGeneReviewUuidMock).toHaveBeenNthCalledWith(1, 'ABL1', 'fake-uuid', true);
+    });
+
+    it('should not update lastReviewed when it is present', async () => {
+      const store = new FirebaseReviewableCrudStore<any>(rootStore);
+      store.data = { key: 'new value', key_uuid: 'fake-uuid', key_review: { lastReviewed: 'original' } };
+
+      await store.updateReviewableContent('Gene/ABL1', 'key', 'new value 2');
+
+      expect(mockUpdate).toHaveBeenCalled();
+
+      // lastReviewed should not be updated because we it was already set
+      const updateValueArg = mockUpdate.mock.calls[0][1];
+      expect(updateValueArg).toBeDefined();
+      expect(updateValueArg).toEqual({
+        key: 'new value 2',
+        key_review: { updateTime: defaultDate.getTime(), updatedBy: 'test user', lastReviewed: 'original' },
+      });
+    });
+
+    it('should remove lastReviewed when value is reverted to original', async () => {
```

Comment:
I actually did not think about this scenario and I should definitely update the logic in the reviewable crud store.

For this case, do we not want to attach `_review`? I see that in `Review` we may have the property `added`, which I assume is for this scenario? If the content was never reviewed, then any changes we make now are considered `added`.

So the final review object should look like:
```
field: 'new value',
field_review: {
  updateTime: 10000000;
  updatedBy = 'test user';
  added: true
}
```

Reply:
Good call on the added. Please create a test for it.

Reply:
In the old curation platform, we set `lastReviewed` to empty string when the content has never been reviewed before. We will keep the same behavior here and I added some tests for when the original content is an empty string.

The `added` property is only added to `name_review`, when a new Gene, Mutation, etc is created. Also the `removed` property is added to the `name_review` when an entity is removed. Since I don't have these functionalities in this PR, I include the tests when we've added those features.